src/server/registry/manifest/manifest.go

```diff
@@ -56,42 +54,59 @@ func (h *handler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 	}
 }
 
-// make sure the artifact exist before proxying the request to the backend registry
-func (h *handler) head(w http.ResponseWriter, req *http.Request) {
-	// TODO check the existence
-	h.proxy.ServeHTTP(w, req)
-}
-
 // make sure the artifact exist before proxying the request to the backend registry
 func (h *handler) get(w http.ResponseWriter, req *http.Request) {
-	// TODO check the existence
+	// check the existence in the database first
+	vars := mux.Vars(req)
+	reference := vars["reference"]
+	artifact, err := h.artCtl.GetByReference(req.Context(), vars["name"], reference, nil)
+	if err != nil {
+		error.Handle(w, req, err)
```

Comment:
This method can be wrapped as the field method of the handler to avoid passing so many params

Reply:
You mean something like that `func (h *handler) sendError()`? But the params `w`, `req` and `err` are all not the fields of handler

Anyway, we'll refine the way to handle error in all handlers, issue https://github.com/goharbor/harbor/issues/10553 is opened to track this

Reply:
Should have such method in the structure who has rw and r